### PR TITLE
Fix user deletion with archived meetings

### LINKED ISSUE #1792: Issue with participant deletion
**Description:** When an participant is deleted from an active meeting, he or she is also incorrectly deleted from an archived meeting in which he or she participated. With this deletion, all references to the participant disappear in the archived meeting and in the account list.  

**Reproduction:**
Precondition: Two meetings are needed. One is archived.  Participant list on both meetings should include at least one identical account. 
1. Open active meeting
2. go to Participants > delete the user who is part of the archived meeting
3. open archived meeting and check the participant list. 

**What should happen:**
If a user is part in an archived meeting, the deletion should only takes part in the active meeting, where he or she is deleted from and not global. The deletion rule for archived meeting should be the same like for active meetings.

### PR DESCRIPTION
**Pull request by jsangmeister**
fixes #1792 

Changed the user scope calculation to also take archived meetings into account, since meeting admins should not be allowed to delete users which are in their meeting plus in other archived meetings.

### CODE REVIEW

**Review by r-peschke — CHANGES_REQUESTED** [review]
- I'm not convinced that we should change the user scope in this case, but that's to discuss.
- Please fix the checks
- if we do it this way, change the doc in wiki

**Comment by jsangmeister** [comment]
Updated the wiki: https://github.com/OpenSlides/OpenSlides/wiki/Users did not mention archived meetings anyway and https://github.com/OpenSlides/OpenSlides/wiki/Archived-Meetings had one hint, which I removed. Also fixed the tests.

**Comment by jsangmeister** [comment]
@r-peschke if you update the priority of the PR, you also have to update the priority of the issue. The `critical` evaluation came from @MSoeb, I just copied it from the issue

**Comment by r-peschke** [comment]
Wrote comment into the issue as answer to the problem description.

**Comment by jsangmeister** [comment]
Reverted the wiki changes. Will update this PR with the changes discussed in https://github.com/OpenSlides/openslides-backend/issues/1795 and https://github.com/OpenSlides/openslides-backend/issues/1792

**Comment by jsangmeister** [comment]
Updated the PR with the presenter changes. Also finally implemented the improvement suggested in https://github.com/OpenSlides/openslides-backend/issues/1213 to reduce the amount of filter requests (3 instead of 3 times the amount of meetings for each user).

File: tests/system/presenter/test_get_user_scope.py [hunk 1]
```diff
@@ -11,22 +11,22 @@ def test_good(self) -> None:
...
             {
                 "meeting/1": {
-                    "name": "meeting1",
                     "committee_id": 2,
                     "is_active_in_organization_id": 1,
                 },
+                # archived meeting
+                "meeting/2": {
+                    "committee_id": 2,
+                    "is_active_in_organization_id": None,
```
Comment (r-peschke) [inline tests/system/presenter/test_get_user_scope.py:20]: To be correct `is_archived_in_organization_id` is necessary to determine an archived meeting

File: tests/system/presenter/test_get_user_related_models.py [hunk 2]
```diff
@@ -173,24 +183,72 @@ def test_get_user_related_models_no_permissions(self) -> None:
...
+            "2": {
+                "meetings": [
+                    {
+                        "id": 1,
+                        "name": "test",
+                        "is_active_in_organization_id": None,
+                    }
+                ]
+            },
+        }
```
Comment (r-peschke) [inline tests/system/presenter/test_get_user_related_models.py:251]: The result still misses the `organization_management_level: OML-String,` info, which is mentioned in the presenter doc. The existing test **test_get_user_related_models_permissions** , beginning at line 279, should have such a result entry, but it is missed and not checked.

**Review by r-peschke — APPROVED** [review]
Just some smaller changes

**Review by r-peschke — CHANGES_REQUESTED** [review]
See 3 small changes in the comments

File: tests/system/presenter/test_get_user_related_models.py (hunk 2 above)
Comment (jsangmeister) [reply tests/system/presenter/test_get_user_related_models.py:264]: Somehow I thought that the presenter already returned the OML. Added it now

**Comment by jsangmeister** [comment]
Added the OML, fixed the tests & comments and reduced the amount of datastore requests in the presenter. Review again @r-peschke

**Review by r-peschke — APPROVED** [review]

**Comment by jsangmeister** [comment]
This still needs changes in the client as described in https://github.com/OpenSlides/openslides-backend/issues/1792#issuecomment-1623203037

**Pull request merged**